### ISSUE TITLE
Move off of PAPR

### ISSUE THREAD

**Issue by jlebon** [containers/bubblewrap#348]
Hi folks,

This repo is now the final one to be using PAPR. I'd like to propose moving off it very soon. But I don't want to just turn it off without a replacement in place. Seems like the easiest path is to match the rest of the `containers/` org and move to Cirrus CI? Otherwise, we can also pretty easily hook up bubblewrap to CoreOS CI (see https://github.com/coreos/fedora-coreos-tracker/issues/263 and https://github.com/jlebon/coreos-ci) (or we might want to do that regardless to run a subset of rpm-ostree tests at least).

/cc @cevich

**Comment by jlebon** [comment]
Probably also worth mentioning GitHub Actions, which Flatpak has migrated to as well.

**Comment by jlebon** [comment]
Aaand... also just found https://github.com/containers/bubblewrap/issues/335, which is about migrating to Prow instead.

So to sum up the options (and those are not necessarily mutually exclusive as mentioned above):
- Cirrus CI (in use by a bunch of other repos in the containers org)
- CoreOS CI (in use by many repos in the coreos org, including rpm-ostree)
- GitHub Actions (in use by flatpak)
- Prow (also in use by rpm-ostree for sanity checks)

Given that the primary users of bubblewrap are flatpak and rpm-ostree, maybe a combination of GitHub Actions and CoreOS CI? That way we should be able to re-use those tests without too much pain.

**Comment by jlebon** [comment]
Any thoughts on this?

**Comment by cevich** [comment]
I might actually consider suggesting dropping Cirrus-CI from the "possible" list:   IIRC, this project would really benefit from testing in an actual atomic-host-like environment, no?  We got away from that requirement in other projects, where it was more acceptable to just use a full-blown Fedora VM but run everything in a container.  I'm not sure if that makes sense here or not, it seems like not.  Most of this is due to the difficulty of running the google-cloud daemons on an atomic-like OS in a way which plays nicely with Cirrus (I tried, it was very hard).  Maybe it could be made to work on Cirrus-CI with another supported cloud, like AWS or Azure, but I have no experience with those.

I also would not recommend Github Actions yet.  Without a custom runner it's pretty well locked in to Ubuntu on the host-side without any possibility to escape.  It's also not had enough public-use cooking time for some basic wrinkles to be worked out.  I'm hopeful it will improve in a year or so, but possibly never for this repo's use-case.

IMHO, CoreOS CI seems the most reasonable (though, I literally know nothing about it, beyond the name) :smile:
